### PR TITLE
Added TabletServer in WaitForBalanceIT

### PR DESCRIPTION
**Pull request by dlmarion**
WaitForBalanceIT was assuming that the tablets were not balanced after adding splits. In the last step of the split code, in DeleteOperationIds, the EventCoordinator is notified to try and get the new splits hosted as fast as possible. This seemed to have created a race condition in the test. Instead, this commit adds a new TabletServer which will cause a balancing to occur. I also modified the isBalanced method to take into account the tservers in the cluster, not just the location from the metadata.

### CODE REVIEW

**Comment by keith-turner** [comment]
> . In the last step of the split code, in DeleteOperationIds, the EventCoordinator is notified to try and get the new splits hosted as fast as possible.

That makes me wonder if the current algorithm the manager runs for wait for balance is flawed.  Wondering if the algorithm should wait for the metadata table to be settled before it waits for the balancer to run.  Not completely sure about this, just trying to improve the handling of that situation.


```java
// scans the metadata table and counts the number tablets that do not have 
// a location and should OR have a location and should not (like offline tables)
public int countMissassignedTablets() {

}

public void waitForBalance(){

          while(countMissassignedTablets() != 0){
              // TODO sleep using a retry obj
          }
    
          waitForBalancerToRun();
}

```

**Comment by dlmarion** [comment]
The test code looks like the following in `main`:
```
      c.tableOperations().addSplits(tableName, partitionKeys);
      assertFalse(isBalanced(c));
      c.instanceOperations().waitForBalance();
```
The failure is in the assertion in the middle line, not the call to `waitForBalance`. `isBalanced()` is a method in the test that checks the assignments in the metadata table, and it's returning `true`. So between the time the `addSplits` call returns and `isBalanced` performs its checks, the Manager has balanced the newly added tablets.

**Comment by keith-turner** [comment]
> So between the time the addSplits call returns and isBalanced performs its checks, the Manager has balanced the newly added tablets.

Seems like the assertFalse should be dropped, its always going to have the potential for race conditions.  Seems the test should do something like the following.

```java
      // should cause the table to become unbalanced
      c.tableOperations().addSplits(tableName, partitionKeys);
      c.instanceOperations().waitForBalance();
      assertTrue(isBalanced(c));
```

**Comment by dlmarion** [comment]
Adding another tserver seems like a valid resolution also. Is one better than the other?

**Comment by keith-turner** [comment]
> Adding another tserver seems like a valid resolution also. Is one better than the other?

Adding the tserver seems fine.  I was just thinking that no matter what we do, the assertFalse prior to the waitForBalance call will always have race conditions.  Adding a tserver lowers the probability of that happening.  Seems liike we could add the tserver or remove the assertFalse call.  Conceptually we know adding splits should disrupt things if there are >1 tservers. The test should probably have asserted there were at least 2 tsevers, it was not checking the number of tservers until these changes checked for 3 tservers.  

Nothing to do with these changes, in general I am not sure the existing test is actually doing anything useful.  The assertFalse call will always have race conditions, then after the call to waitForBalance() when the system should be balanced the test code waits for balance on its own.

```java
      c.instanceOperations().waitForBalance();
     // If at this point the system is not balanced the test can still pass and does not care, so what is it testing?
      Wait.waitFor(() -> isBalanced(c));
```

We could make changes to get the test to pass for now and open a follow on issue about investigating general improvements to the test.  Maybe ideally the test should look like the following.

```java
      c.tableOperations().addSplits(tableName, partitionKeys);
      assertTrue(numTservers() >= 2);
      c.instanceOperations().waitForBalance();
      assertTrue(isBalanced(c));
```


If adding the tserver gets the test passing for now, that seems like an ok short term fix.

**Comment by dlmarion** [comment]
I think I merged our ideas in 2bb8649

**Comment by dlmarion** [comment]
Not sure what's happing with GitHub Actions right now.... It's building fine.

File: test/src/main/java/org/apache/accumulo/test/WaitForBalanceIT.java [hunk 1]
```diff
@@ -115,13 +126,13 @@ private boolean isBalanced(AccumuloClient c) throws Exception {
       return false;
     }
     int average = 0;
-    for (Integer i : counts.values()) {
+    for (Integer i : tserverCounts.values()) {
       average += i;
     }
-    average /= counts.size();
-    System.out.println(counts);
+    average /= tserverCounts.size();
```
Comment (keith-turner) [inline test/src/main/java/org/apache/accumulo/test/WaitForBalanceIT.java:132]: Computing the average seems like it should use a double.  This is an existing problem and not sure if even matters.

**Review by keith-turner — APPROVED** [review]

**Pull request merged**